lib/mapObjects/CGTownInstance.h

```diff
@@ -73,6 +73,9 @@ class DLL_LINKAGE CGTownInstance : public CGDwelling, public IShipyard, public I
 	std::vector<std::vector<SpellID> > spells; //spells[level] -> vector of spells, first will be available in guild
 	std::vector<CCastleEvent> events;
 	std::pair<si32, si32> bonusValue;//var to store town bonuses (rampart = resources from mystic pond, factory = save debts);
+	std::map<int, int> spellResearchActionsPerDay;
+	int spellResearchCounter;
```

Comment:
Why do you need `spellResearchActionsPerDay`?
If you need "how many times spell research was done today" then you can use simple counter and reset it in `NewTurn::applyGs`, similar to built buildings counter

Reply:
Ah thanks. This was the missing puzzle piece. :)